### PR TITLE
M1 mac build support in maven deps

### LINKED ISSUE #2073: Build fails on M1 macs due to unavailable binaries
Some of the grpc build dependencies, for example protoc, use binary dependencies. We need to bump some of the versions so we have binaries available for the aarch64 architecture used on newer macs.

Building the v2.0.0 branch currently fails when looking for the protoc binary in maven central:

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  23.570 s
[INFO] Finished at: 2022-09-15T10:35:57+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.xolstice.maven.plugins:protobuf-maven-plugin:0.6.1:compile (default) on project grpc-proto: Unable to resolve artifact: Missing:
[ERROR] ----------
[ERROR] 1) com.google.protobuf:protoc:exe:osx-aarch_64:3.12.0
[ERROR]
[ERROR]   Try downloading the file manually from the project website.
[ERROR]
[ERROR]   Then, install it using the command:
[ERROR]       mvn install:install-file -DgroupId=com.google.protobuf -DartifactId=protoc -Dversion=3.12.0 -Dclassifier=osx-aarch_64 -Dpackaging=exe -Dfile=/path/to/file
[ERROR]
[ERROR]   Alternatively, if you host your own repository you can deploy the file there:
[ERROR]       mvn deploy:deploy-file -DgroupId=com.google.protobuf -DartifactId=protoc -Dversion=3.12.0 -Dclassifier=osx-aarch_64 -Dpackaging=exe -Dfile=/path/to/file -Durl=[url] -DrepositoryId=[id]
[ERROR]
[ERROR]   Path to dependency:
[ERROR]   	1) io.stargate.grpc:grpc-proto:jar:2.0.0-ALPHA-5-SNAPSHOT
[ERROR]   	2) com.google.protobuf:protoc:exe:osx-aarch_64:3.12.0
[ERROR]
[ERROR] ----------
[ERROR] 1 required artifact is missing.
[ERROR]
[ERROR] for artifact:
[ERROR]   io.stargate.grpc:grpc-proto:jar:2.0.0-ALPHA-5-SNAPSHOT
[ERROR]
[ERROR] from the specified remote repositories:
[ERROR]   yelp-nexus (https://nexus.yelpcorp.com/nexus/content/groups/public, releases=true, snapshots=true),
[ERROR]   central (https://jcenter.bintray.com, releases=true, snapshots=false)
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :grpc-proto
```

### PR DESCRIPTION
**Pull request by tub**
**What this PR does**:
Bumping binary dependencies to versions where there's aarch64 binary 

Locally the `testing` module still fails with 

```
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:23 min
[INFO] Finished at: 2022-09-15T10:32:23+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project testing: Compilation failure: Compilation failure:
[ERROR] /Users/tobyc/IdeaProjects/datastax-stargate/testing/src/main/java/io/stargate/it/grpc/streaming/ExecuteQueryStreamingTest.java:[33,41] cannot find symbol
[ERROR]   symbol:   class StreamingResponse
[ERROR]   location: class io.stargate.proto.QueryOuterClass
[ERROR] /Users/tobyc/IdeaProjects/datastax-stargate/testing/src/main/java/io/stargate/it/grpc/streaming/ExecuteBatchStreamingTest.java:[35,41] cannot find symbol
[ERROR]   symbol:   class StreamingResponse
[ERROR]   location: class io.stargate.proto.QueryOuterClass
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :testing
```
However this is a lot further than it got before.

**Which issue(s) this PR fixes**:
Fixes #2073 

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by maheshrajamani** [comment]
Hi @tub, 
There is code conflict, can you have this resolved?

Thanks
Mahesh

File: pom.xml [hunk 1]
```diff
@@ -490,6 +490,11 @@
   </distributionManagement>
   <dependencyManagement>
     <dependencies>
+      <dependency>
```
Comment (tatu-at-datastax) [inline pom.xml:493]: Could you add a brief comment here wrt why this is needed -- it sounds like it is to override version of a transitive dependency (protobuf generator?); would be good to know wrt future increases of version.

**Comment by tatu-at-datastax** [comment]
Hi @tub! It looks like the PR fails due to reasons not related to changes themselves: I filed #2129 for follow-up.
But re-creating changes as PR #2128 passes tests, so I will merge fix you contributed using that PR.
This is unfortunate but I think we want to get the fix in ASAP and I'd prefer to merge a PR that passed tests (even if test failure is due to unrelated issue).

Thank you again for contributing this fix, and please let us know if there are follow up issues (I have older Mac so I cannot easily verify this working).

**Comment by tatu-at-datastax** [comment]
Oh. Forgot that it's Github action issue, not one for old GCB test issue.

**Comment by tatu-at-datastax** [comment]
PR #2128 merged, closing this as per description above: fix is in `v2.0.0` now.

**Comment by tatu-at-datastax** [comment]
Ah, no problem! And thank you once again; I hope this helps.

**Pull request closed without merging**